syntax = "proto3";

package product;

service ProductService {
  rpc GetProduct(ProductRequest) returns (ProductResponse) {}
  rpc CheckProductAvailability(ProductAvailabilityRequest) returns (ProductAvailabilityResponse) {}
}

message ProductRequest {
  string product_id = 1;
}

message ProductResponse {
  string product_id = 1;
  string title = 2;
  double price = 3;
  int32 stock = 4;
}

message ProductAvailabilityRequest {
  string product_id = 1;
  int32 quantity = 2;
}

message ProductAvailabilityResponse {
  bool available = 1;
  string message = 2;
} 